src/azul/chalice.py

```diff
@@ -504,34 +501,102 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
-    def threshold(self, *, errors: int, throttles: int):
-        def wrapper(f):
+    @attr.frozen(kw_only=True)
+    class LambdaDecorator(ABC):
+        lambda_name: Optional[str] = attr.ib(default=None)
```

Comment:
If you use the new-style API, (`attrs.define`, `attrs.frozen`), this should use `attrs.field`.

This means that you probably first need to convert all usages of the old-style API in this module to the new-style API, as I don't want us to mix old (attr) and new (attrs).

https://www.attrs.org/en/stable/names.html#tl-dr